src/lib/polyfills.ts

```diff
@@ -54,3 +58,33 @@ import '@formatjs/intl-pluralrules/locale-data/tr'
 import '@formatjs/intl-pluralrules/locale-data/uk'
 import '@formatjs/intl-pluralrules/locale-data/vi'
 import '@formatjs/intl-pluralrules/locale-data/zh'
+;(function () {
+  // Ensure global exists
+  if (typeof globalThis !== 'undefined' && typeof global === 'undefined') {
+    ;(globalThis as any).global = globalThis
+  }
+
+  // Preserve existing process.env from vite's define config
+  const existingEnv = typeof process !== 'undefined' && process.env ? process.env : {}
+
+  // Create process object if it doesn't exist
+  if (typeof process === 'undefined') {
+    ;(globalThis as any).process = { env: existingEnv, versions: {} }
+  }
+
+  // Add nextTick if missing
+  if (!process.nextTick) {
+    // Modern approach: direct queueMicrotask or Promise fallback
+    // This provides proper microtask timing that Node.js code expects
+    process.nextTick = function (callback: Function, ...args: any[]) {
+      const fn = () => callback(...args)
+
+      if (typeof queueMicrotask === 'function') {
+        queueMicrotask(fn)
+      } else {
+        // Fallback for older browsers
+        Promise.resolve().then(fn)
```

Comment:
Yeah, that's ugly, not sure whether users will hit this.
Promise.resolve automagically gets queued in the event loop as next tick, so, yeah, more or less the same.
https://stackoverflow.com/questions/27647742/promise-resolve-then-vs-setimmediate-vs-nexttick